src/tests/test_set_based_verifier.cpp

```diff
@@ -0,0 +1,202 @@
+        : params_map_({{config::names::kCsvConfig, csv_config},
+                       {config::names::kOcContext, oc_context},
+                       {config::names::kOcLeftIndex, oc_left},
+                       {config::names::kOcRightIndex, oc_right},
+                       {config::names::kOcLeftOrdering, ordering},
+                       {config::names::kOFDContext, ofd_context},
+                       {config::names::kOFDRightIndex, ofd_right}}),
+          expected_removal_set_(removal_set) {}
+
+    algos::StdParamsMap const& GetParamsMap() const {
+        return params_map_;
+    }
+
+    RemovalSetAsVec const& GetExpectedRemovalSet() const {
+        return expected_removal_set_;
+    }
+};
+
+class TestSetBasedAodVerifier : public ::testing::TestWithParam<SetBasedAodVerifierParams> {};
+
+TEST_P(TestSetBasedAodVerifier, CorrectnessTest) {
+    SetBasedAodVerifierParams const& params(GetParam());
+    algos::StdParamsMap const& params_map = params.GetParamsMap();
+    auto verifier = algos::CreateAndLoadAlgorithm<SetBasedAodVerifier>(params_map);
+    verifier->Execute();
+    EXPECT_EQ(verifier->Holds(), params.GetExpectedRemovalSet().empty());
+
+    RemovalSet actual_removal_set = verifier->GetRemovalSet();
+    RemovalSetAsVec actual_removal_set_vec(actual_removal_set.begin(), actual_removal_set.end());
+    std::sort(actual_removal_set_vec.begin(), actual_removal_set_vec.end());
+    RemovalSetAsVec expected_removal_set_vec = params.GetExpectedRemovalSet();
+    std::sort(expected_removal_set_vec.begin(), expected_removal_set_vec.end());
+    EXPECT_EQ(actual_removal_set_vec, expected_removal_set_vec);
+    SetBasedAodVerifier::Error expected_error =
+            static_cast<SetBasedAodVerifier::Error>(expected_removal_set_vec.size()) /
+            verifier->GetTupleCount();
+    EXPECT_EQ(verifier->GetError(), expected_error);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+        SetBasedAodVerifierTestSuite, TestSetBasedAodVerifier,
+        ::testing::Values(/* {}: Col5 -> Col1 */
+                          SetBasedAodVerifierParams(kTestFD, /*oc_context=*/{},
+                                                    /*oc_left=*/5, /*oc_right=*/1,
+                                                    /*ordering=*/Ordering::ascending,
+                                                    /*ofd_context=*/{5}, /*ofd_right=*/1,
+                                                    /*removal_set=*/{}),
+                          /* {}: Col5 -> Col2 */
+                          SetBasedAodVerifierParams(kTestFD, /*oc_context=*/{},
+                                                    /*oc_left=*/5, /*oc_right=*/2,
+                                                    /*ordering=*/Ordering::ascending,
+                                                    /*ofd_context=*/{5}, /*ofd_right=*/2,
+                                                    /*removal_set=*/{2, 5, 8}),
+                          /* {Col3}: Col4 -> Col1 */
```

Comment:
Idk if this is possible, but can we add some descriptive names to individual tests in this test suite? Like make those comments `{}: Col5 -> Col1` into names? If it's hard, then feel free to ignore